Set Calendar time zone

apooravc opened the pull request:
This PR adds a global `calendar_timezone` which the user could use to select his/her time zone for Calendar date/time functions. [date_default_timezone_get](http://php.net/manual/en/function.date-default-timezone-get.php) (used as default time zone for Calendar) gets the value of `date.timezone` in php.ini file if it is valid/supported (valid time zones given [here](http://php.net/manual/en/timezones.php)) otherwise defaults to UTC. The user selected time zone from Main globals or User specific globals (given more preference when different values) is set as `date.timezone` configuration option value using [ini_set](http://php.net/manual/en/function.ini-set.php). 

`date` function is used to fill entries in `date` & `start_datetime` columns in patient_tracker and patient_tracker_element tables respectively. @teryhill Yet to add other time zones ([list here](http://php.net/manual/en/timezones.php)). Should I add all of which are given there? Or go for a particular set covering major regions in time zones?

teryhill commented:
look at the demo files at full calendar. get-timezones.php, get-events.php and utils.php.

aethelwulffe commented on library/globals.inc.php:
```diff
@@ -1537,6 +1538,19 @@
+      array(
+        date_default_timezone_get() => xl('Default - php.ini value'),
+        'Asia/Dubai' => xl('Dubai'),
+        'Asia/Kolkata' => xl('Kolkata'),
```
I knew that! :)

apooravc commented:
In the third commit:

1) Moved global check code from `library/patient_tracker.inc.php` to `interface/globals.php` to cover time zones all over EHR.

2) Added reference to `globals.php` from `patient_tracker.inc.php` file.

Note about `ini_set` function: "Sets the value of the given configuration option. The configuration option will keep this new value during the _script's execution_, and will be restored at the script's ending." Referring here to those scripts which will use `date` functions which depend on time zones. Needs to be tested wherever time is involved. @teryhill Please have a look.

aethelwulffe commented:
Good idea.  This stuff has been on todo lists for a while.  One place for time stuff, and only one place.

apooravc commented:
@teryhill Also, this PR was first intended to be used to set time zone for Calendar only but now it can be applied to whole of EHR. So, should I let global option for setting the time zone remain under Calendar sub-section or should I shift it to somewhere else (and if shifting, then where)?

teryhill commented:
Move it to under the Systems tab

teryhill commented:
Also look at moving your function from globals.inc.php to edit_globals.php.

apooravc commented:
@teryhill The code creating `$zone_opt_arr` right? Is it causing errors?

teryhill commented:
@apooravc it is not causing errors. It is just "Cleaner" to have it outside of globals.

teryhill commented on library/patient_tracker.inc.php:
```diff
@@ -19,6 +19,7 @@
 */
 require_once(dirname(__FILE__) . '/appointments.inc.php');
+require_once("../interface/globals.php");
 
```
is this needed?

apooravc replied:
Yes, the code in globals.php uses `ini_set` set the selected time zone in Globals as time zone for the script in which globals.php is included. So, wherever date/time functions, this should be included in script containing them.

teryhill commented:
It is called in a program that includes this one.

apooravc commented:
@teryhill You mean that globals.php is called in edit_globals? Yes, but edit_globals only lets a user save his/her global option which could be then accessed by `$GLOBALS['calendar_timezone']`. Suppose we're saving an appointment, then `patient_tracker.inc.php` is executed too and without including globals.php here, the time zone for `date` function inside `manage_tracker_status` would not be saved according to user saved option.

teryhill commented:
The time zone information is in globals.php, correct. Globals php is a main program that is loaded with the main code . Look at main.php, it is loaded there, look at patient_tracker.php , it is loaded there. patient_tracker.inc is used by patient tracker. Globals is called by the "Top" programs.

apooravc commented:
Changes done in last two commits:

1. Removed `include globals.php` from patient_tracker.inc.php as it is included in `main.php` itself.
2. Changed global name from `calendar_timezone` to `ehr_timezone` as it now covers time zone all over EHR.
3. Moved time zone option list from Calendar to System tab in Admin > Globals and also moved code from `globals.inc.php` to `edit_globals.php`.

@teryhill Please have a look.

teryhill commented:
This option is a user specific item also @aethelwulffe so if the user is in another time zone they can set there system for the different zone. Needs to have the / put back in so the locations are easier to determine.

apooravc commented:
@teryhill Added back `/` to options so that user may find it easier to navigate to a time zone. One more thing I see is that after moving global from Calendar to System tab, the time zone option list now only appears in Admin > Globals > System and not also in User Preferences.

teryhill commented:
You need to add the "Systems" tag to the users.

![image](https://user-images.githubusercontent.com/4655366/41731875-8e8dbb6c-754d-11e8-9e1d-0c461b92588b.png)

apooravc commented:
@teryhill Thought so, on it.

apooravc commented:
@teryhill Added System tab to Edit User Settings. Please have a look.

apooravc commented:
@aethelwulffe These changes are final.
